docs/developer/design/volume_v1.0.rst

```diff
@@ -0,0 +1,298 @@
+  nodes
+* update the Platform UI to allow volume management
+
+
+User Stories
+^^^^^^^^^^^^
+
+
+Volume Creation
+~~~~~~~~~~~~~~~
+
+As a user I need to be able to create MetalK8s volume from the Platform UI.
+
+At creation time I can specify the type of volume I want, and then either its
+size (for **sparseLoopDevice**) or the backing device (for **rawBlockDevice**).
+
+I should be able monitor the progression of the volume creation from the
+Platform UI and see when the volume is ready to use (or if an error occured).
+
+
+Volume Monitoring
+~~~~~~~~~~~~~~~~~
+
+As a user I should be able to see all the volumes existing on a specified node
+as well as their states.
+
+
+Volume Deletion
+~~~~~~~~~~~~~~~
+
+As a user I need to be able to delete a MetalK8s volume from the Platform UI
+when I no longer need it.
+
+The Platform UI should prevent me to delete a used volume.
+
+I should be able monitor the progression of the volume deletion from the
+Platform UI.
+
+
+Components Interactions
+^^^^^^^^^^^^^^^^^^^^^^^
+
+User will create Metalk8s volumes through the Platform UI.
+
+The Platform UI will create and delete **Volume** CRs from the API server.
+
+The operator will watch events related to **Volume** CRs and
+**PersistentVolume** owned by **Volume** and react in order to update the state
+of the cluster to meet the desired state (prepare storage when a new **Volume**
+CR is created, clean up resources when a **Volume** CR is deleted). It will
+also be responsible for updating the states of the volumes.
+
+To do its job, the operator will rely on Salt states that will be called
+asynchronously (to avoid blocking the reconciliation loop and keep a reactive
+system) through the Salt API. Authentication to the Salt API will be done
+though a dedicated Salt account (with limited privileges) using credentials
+from a dedicated cluster **Service Account**.
+
+.. uml:: volume_v1.0-creation_seqdiag.uml
+
+.. uml:: volume_v1.0-deletion_seqdiag.uml
+
+
+Implementation Details
+----------------------
+
+
+Volume Status
+^^^^^^^^^^^^^
+
+A **PersistentVolume** from Kubernetes has the following states:
+
+* **Pending**: used for **PersistentVolume** that is not available
+* **Available**: a free resource that is not yet bound to a claim
+* **Bound**: the volume is bound to a claim
+* **Released**: the claim has been deleted, but the resource is not yet
+  reclaimed by the cluster
+* **Failed**: the volume has failed its automatic reclamation
+
+Similarly, our **Volume** object will have the following states:
+
+* **Available**: the backing storage is ready and the associated
+  **PersistentVolume** was created
+* **Pending**: preparation of the backing storage in progress (e.g.
+  an asynchronous Salt call is still running).
+* **Failed**: something is wrong with the volume (Salt state execution failed,
+  invalid value in the CRD, …)
+* **Terminating**: cleanup of the backing storage in progress (e.g.
```

Comment:
Shouldn't this be `i.e.`?

Reply:
No, because that's not the only case (but the most common one).

You enter the `Terminating` state once you do the Salt call, but after the call you have other steps such as deleting the finalizers, if you have network issues this may need retries and meanwhile you'll stay in `Terminating` state.